src/mscorlib/shared/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -692,6 +689,35 @@ protected EventSource(EventSourceSettings settings, params string[] traits)
             Initialize(eventSourceGuid, eventSourceName, traits);
         }
 
+#if FEATURE_PERFTRACING
+        // Generate all EventPipeEvent instances which belong to this EventProvider
+        private unsafe void GenerateEventPipeEvent()
+        {
+            List<uint> allEvents = new List<uint>();
+            Debug.Assert(m_eventData != null);
+            int cnt = m_eventData.Length;
+            for (int i = 0; i < cnt; i++)
+            {
+                Int64 keywords = m_eventData[i].Descriptor.Keywords;
+                uint eventID = (uint)m_eventData[i].Descriptor.EventId;
+                uint eventVersion = m_eventData[i].Descriptor.Version;
+                uint level = m_eventData[i].Descriptor.Level;
+                // TODO: what's the value of needStack?
+                bool needStack = false;
```

Comment:
needStack is not a attribute of an event (it can be removed).   Basically any event can have a stack and it is up to the consumer to decide (as long as you are still on the thread, you can take a stacks later).  

I would probably simply drop needStack.